druid-shell/src/screen.rs

```diff
@@ -0,0 +1,94 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Module to get information about monitors
+
+use crate::kurbo::{Rect, Size};
+use crate::platform;
+use std::fmt;
+use std::fmt::Display;
+
+/// Monitor struct containing data about a monitor on the system
+///
+/// Use Screen::get_monitors() to return a Vec<Monitor> of all the monitors on the system
+#[derive(Clone)]
+pub struct Monitor {
+    primary: bool,
+    rect: Rect,
+    // TODO: Work area, cross_platform
+    // https://developer.apple.com/documentation/appkit/nsscreen/1388369-visibleframe
+    // https://developer.gnome.org/gdk3/stable/GdkMonitor.html#gdk-monitor-get-workarea
+    // https://docs.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-monitorinfo
+    // Unsure about x11
+    work_rect: Rect,
+}
+
+impl Monitor {
+    #[allow(dead_code)]
+    pub(crate) fn new(primary: bool, rect: Rect, work_rect: Rect) -> Self {
+        Monitor {
+            primary,
+            rect,
+            work_rect,
+        }
+    }
+    /// Returns true if the monitor is the primary monitor.
+    /// The primary monitor has its origin at (0, 0) in virtual screen coordinates.
+    pub fn is_primary(&self) -> bool {
+        self.primary
+    }
+    /// Returns the monitor rectangle in virtual screen coordinates.
+    pub fn virtual_rect(&self) -> Rect {
+        self.rect
+    }
+
+    /// Returns the monitor working rectangle in virtual screen coordinates.
+    /// The working rectangle excludes certain things like the dock and menubar on mac,
+    /// and the taskbar on windows.
+    pub fn virtual_work_rect(&self) -> Rect {
+        self.work_rect
+    }
+}
+
+impl Display for Monitor {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.primary {
+            write!(f, "Primary ")?;
+        } else {
+            write!(f, "Secondary ")?;
+        }
+        write!(
+            f,
+            "({}, {})({}, {})",
+            self.rect.x0, self.rect.x1, self.rect.y0, self.rect.y1
+        )?;
+        Ok(())
+    }
+}
+
+/// Information about the screen and monitors
+pub struct Screen {}
+impl Screen {
+    /// Returns a vector of all the [`monitors`] on the system.
+    ///
+    /// [`monitors`]: struct.Monitor.html
+    pub fn get_monitors() -> Vec<Monitor> {
+        platform::screen::get_monitors()
+    }
+
+    /// Returns the total virtual screen space on the system, in pixels.
+    pub fn get_display_size() -> Size {
```

Comment:
I think it would make sense for this to be a Rect - the bounding box of all monitors on virtual space. 
On the mac at least you can have screens with origins at negative coordinates vs the 'main screen' which is always at 0, 0 ( bottom left corner as mac coordinates are upside down .. )

Reply:
Just to make sure I understand you correctly, lets say we have a virtual screen where the rect would be
top = -100, left = -100, bottom = 0, right = 0. currently this function would return a Size(100,100) but instead you suggest it should return the Rect(-100, -100, 0, 0) ?

Reply:
I think so - that way you can still get the size (off the rect) but also know the extents of virtual screen coordinates